### PR TITLE
Automation: Label pull requests based on changed files

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**:
Uses https://github.com/grafana/grafana-github-actions/pull/14 and https://github.com/grafana/grafana-github-actions/pull/15 to label pull requests based on changed files.

Having labels on PR's makes it a lot easier to filter out PR's that only targets changes you're responsible for as an example. Docs, frontend, backend, explore and/or certain data sources. Additionally, this is usually a manual task today and could also be used for statistical purposes if we would like to gather statistics about what parts/areas PR's changes.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
- Used the CODEOWNERS config and tried to map that to labels as a first step. 
- Have tested it here https://github.com/grafana/github-actions-testrepo/pull/18

### CODE REVIEW

File: .github/pr-commands.json [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+  },
+  {
+    "type": "changedfiles",
+    "matches": {
+      "any": [
+        "*.go",
+        "go.mod",
+        "go.sum",
+        "/contribute/style-guides/backend.md",
+        "/contribute/architecture/backend"
```
Comment (wbrowne) [inline .github/pr-commands.json:46]: Just an FYI - there's still a few others not in here yet such as`contribute/architecture/services.md` and `contribute/architecture/database.md` for example. But I think this is fine and we can fix those up and leave this as is

**Review by wbrowne — APPROVED** [review]
Amazing work!! A lot less to worry about with this change. 👏 

Excited!

File: .github/pr-commands.json (hunk 1 above)
Comment (wbrowne) [reply .github/pr-commands.json:46]: https://github.com/grafana/grafana/pull/32041

Comment (marefr) [reply .github/pr-commands.json]: Nice thanks. I'll fix

**Review by wbrowne — APPROVED** [review]

**Review by torkelo — APPROVED** [review]
This is looking great! ❤️

**Pull request merged**